runtimes/nodejs/actions/send-message.js

```diff
@@ -399,36 +400,59 @@ function main(params) {
   }
 
   const bodyData = JSON.stringify(sendMessage);
-  let apiHost;
-  if (params.apiHost) {
-    apiHost = params.apiHost;
-  }
-  else if (params.admin_url) {
-    const adminURL = url.parse(params.admin_url).protocol === null ? `https:${params.admin_url}` : params.admin_url;
+  let apiHost = '';
+  if (theParams.apiHost) {
+    apiHost = theParams.apiHost;
+  } else if (theParams.url) {
+    apiHost = url.parse(theParams.url).host;
+  } else if (theParams.admin_url) {
+    const adminURL = url.parse(theParams.admin_url).protocol === null ? `https:${theParams.admin_url}` : theParams.admin_url;
     apiHost = url.parse(adminURL).host;
   } else {
     apiHost = 'mobile.ng.bluemix.net';
   }
+  let header;
+  let pushHeaders = {};
+  if (isIamAuth) {
+    try {
+      header = await handleAuth(theParams.apikey)
+    } catch (err) {
+      return Promise.reject({
+        statusCode: 500,
+        headers: { 'Content-Type': 'application/json' },
+        body: { message: 'Error processing your request' },
+      });
+    }
+    pushHeaders = {
+      Authorization: header,
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    };
+  } else {
+    pushHeaders = {
+      appSecret,
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    };
+  }
 
-  const promise = new Promise((resolve, reject) => {
-    request({
+  let sendMessageResult;
+  try {
+    sendMessageResult = await (request({
       method: 'post',
       uri: `https://${apiHost}/imfpush/v1/apps/${appId}/messages`,
-      headers: {
-        appSecret,
-        Accept: 'application/json',
-        'Content-Type': 'application/json',
-      },
+      headers: pushHeaders,
       body: bodyData,
-    }, (error, response, body) => {
-      if (error) {
-        reject(error);
-      }
-      const j = JSON.parse(body);
-      resolve(j);
-    });
-  });
-  return promise;
+    }));
+  } catch (err) {
+    return Promise.reject({
+      statusCode: 500,
+      headers: { 'Content-Type': 'application/json' },
+      body: { message: err },
+    })
+  }
+  const j = JSON.parse(sendMessageResult.body);
```

Comment:
seems like this should be in try ... catch block

Reply:
yeah, you're right.  The original code didn't have it in there either.